ivy/functional/frontends/torch/linalg.py

```diff
@@ -94,3 +94,33 @@ def svd(A, /, *, full_matrices=True, driver=None, out=None):
 def svdvals(A, *, driver=None, out=None):
     # TODO: add handling for driver
     return ivy.svdvals(A, out=out)
+
+
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"1.11.0 and below": ("float16", "bfloat16")}, "torch")
+def tensorinv(input, ind=2):
+    input_shape = ivy.shape(input)
+    if ind > 0:
+        shape_ind_end = input_shape[:ind]
+        shape_ind_start = input_shape[ind:]
+        prod_ind_end = 1
+        prod_ind_start = 1
+        for i in shape_ind_start:
+            prod_ind_start *= i
+        for j in shape_ind_end:
+            prod_ind_end *= j
+        if prod_ind_end == prod_ind_start:
+            inverse_shape = shape_ind_start + shape_ind_end
+            input = ivy.reshape(input, shape=(prod_ind_end, -1))
+            inverse_shape_tuple = tuple([*inverse_shape])
+            if len(ivy.shape(input)) > 1:
+                inverse_tensor = ivy.inv(input)
+            else:
```

Comment:
Does the original `tensorinv` function work for 1-dimensional input arrays? Maybe you should raise an error instead of returned a reshaped array in that case.

Reply:
![clion64_9f7jPFEAqJ](https://user-images.githubusercontent.com/51514616/209119340-32326209-6ead-4ae1-b31e-c202cbeb527e.png)
![clion64_Tf8Q7y6hXd](https://user-images.githubusercontent.com/51514616/209119350-c4f4f392-c70c-44bb-9628-23317bf9000c.png)
I tried to follow logic implemented at origin of the function.  I also checked jax version of it. Here original function has no limit for case `ind=1` as far as i see.

Reply:
Then, you should not handle 1-d arguments either. How did you exclude them from the test and got an error?

Reply:
I set min dim size to 2:
```py
def _get_inv_square_matrices(draw):
    dim_size = draw(helpers.ints(min_value=2, max_value=10))
```
and got this error:
`hypothesis.errors.FailedHealthCheck: The smallest natural example for your test is extremely large.`

It is probably related to the wrong helper I used. I might solve this by removing that helper and using one you told me. I'll update when I am able to do.